src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.IO;
+using UnityEditor.iOS.Xcode;
+using UnityEditor.iOS.Xcode.Extensions;
+using UnityEngine;
+
+namespace Sentry.Unity.Editor.iOS
+{
+    internal class SentryXcodeProject : IDisposable
+    {
+        private const string FrameworkName = "Sentry.framework";
+
+        private const string MainPath = "MainApp/main.mm";
+        private const string OptionsPath = "MainApp/SentryOptions.m";
+        private string UnityPackageFrameworkRoot = Path.Combine("Frameworks", "io.sentry.unity");
+
+        private readonly string _projectRoot;
+        private readonly SentryUnityOptions _options;
+
+        private readonly PBXProject _project;
+        private readonly string _projectPath;
+
+        internal string? RelativeFrameworkPath { get; set; }
+
+        private readonly INativeMain _nativeMain;
+        private readonly INativeOptions _nativeOptions;
+
+        internal SentryXcodeProject(
+            string projectRoot,
+            SentryUnityOptions options)
+            : this(projectRoot, options, new NativeMain(), new NativeOptions())
+        {
+        }
+
+        internal SentryXcodeProject(
+            string projectRoot,
+            SentryUnityOptions options,
+            INativeMain mainModifier,
+            INativeOptions sentryNativeOptions)
+        {
+            _projectRoot = projectRoot;
+            _options = options;
+
+            _projectPath = PBXProject.GetPBXProjectPath(projectRoot);
+            _project = new PBXProject();
+
+            _nativeMain = mainModifier;
+            _nativeOptions = sentryNativeOptions;
+        }
+
+        public static SentryXcodeProject Open(string path, SentryUnityOptions options)
+        {
+            var xcodeProject = new SentryXcodeProject(path, options);
+            xcodeProject.ReadFromProjectFile();
+            xcodeProject.SetRelativeFrameworkPath();
+
+            return xcodeProject;
+        }
+
+        internal void ReadFromProjectFile()
+        {
+            if (!File.Exists(_projectPath))
+            {
+                throw new FileNotFoundException($"Could not locate generated Xcode project at {_projectPath}");
```

Comment:
Second arg is the file name: https://docs.microsoft.com/en-us/dotnet/api/system.io.filenotfoundexception.-ctor?view=net-5.0#System_IO_FileNotFoundException__ctor_System_String_System_String_


```suggestion
                throw new FileNotFoundException($"Could not locate generated Xcode project", _projectPath);
```